src/components/application_manager/rpc_plugins/app_service_rpc_plugin/test/app_service_plugin/app_service_app_extension_test.cc

```diff
@@ -0,0 +1,220 @@
+  ASSERT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+
+  EXPECT_FALSE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType1));
+  EXPECT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+  const auto& subs = app_service_app_extension->Subscriptions();
+  EXPECT_EQ(1u, subs.size());
+}
+
+TEST_F(
+    AppServiceAppExtensionTest,
+    UnsubscribeFromAppService_AppServiceType1Unsubscribed_AppServiceType2Remains_SUCCESS) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+  // Subscribe
+  ASSERT_TRUE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType1));
+  ASSERT_TRUE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType2));
+  auto subs = app_service_app_extension->Subscriptions();
+  ASSERT_EQ(2u, subs.size());
+  ASSERT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+  ASSERT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType2));
+
+  // Unsubscribe
+  EXPECT_TRUE(
+      app_service_app_extension->UnsubscribeFromAppService(kAppServiceType1));
+  EXPECT_FALSE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+  EXPECT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType2));
+  subs = app_service_app_extension->Subscriptions();
+  EXPECT_EQ(1u, subs.size());
+}
+
+TEST_F(AppServiceAppExtensionTest,
+       UnsubscribeFromAppService_UnsubscribeNotSubscribedAppServiceType_FAIL) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+
+  ASSERT_TRUE(
+      app_service_app_extension->SubscribeToAppService(kAppServiceType1));
+  ASSERT_EQ(1u, app_service_app_extension->Subscriptions().size());
+  ASSERT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+
+  EXPECT_FALSE(
+      app_service_app_extension->UnsubscribeFromAppService(kAppServiceType2));
+  EXPECT_TRUE(
+      app_service_app_extension->IsSubscribedToAppService(kAppServiceType1));
+  EXPECT_EQ(1u, app_service_app_extension->Subscriptions().size());
+}
+
+TEST_F(AppServiceAppExtensionTest,
+       UnsubscribeFromAppService_UnsubscribeAll_SUCCESS) {
+  auto app_service_app_extension =
+      std::make_shared<AppServiceAppExtension>(app_service_plugin_, *mock_app_);
+
+  for (const auto& app_service_type : {kAppServiceType1, kAppServiceType2}) {
+    ASSERT_TRUE(
+        app_service_app_extension->SubscribeToAppService(app_service_type));
+    ASSERT_TRUE(
+        app_service_app_extension->IsSubscribedToAppService(app_service_type));
+  }
+  ASSERT_EQ(2u, app_service_app_extension->Subscriptions().size());
+
+  app_service_app_extension->UnsubscribeFromAppService();
+
+  for (const auto& app_service_type : {kAppServiceType1, kAppServiceType2}) {
```

Comment:
@YarikMamykin please extract this into a separate list and use in both loops

Reply:
@AKalinich-Luxoft fixed in d3a6bc63c0ded3551421683575a2de6a5850a091

Reply:
@AKalinich-Luxoft  fixed in 2ec9ea198cf54b563252051776bd08f48acb560f